standalone-dns-proxy/cmd/standalonednsproxy.go

```diff
@@ -57,8 +57,10 @@ func NewStandaloneDNSProxy(params standaloneDNSProxyParams) *StandaloneDNSProxy
 func (sdp *StandaloneDNSProxy) StartStandaloneDNSProxy() error {
 	sdp.logger.Info("Starting standalone DNS proxy")
 
-	// start the connection handler
-	sdp.connHandler.StartConnection()
+	// watch the connection state and start the DNS proxy once connected
```

Comment:
This function is in the Hive Start hook, which means it is not allowed to take a long time. Connecting may take an unbounded amount of time, as the agent may be down. You'll need to refactor this to be in its own job.

Reply:
I updated it to be a job that watches over the connection.